hudi-common/src/main/java/org/apache/hudi/common/table/timeline/versioning/v1/ArchivedTimelineLoaderV1.java

```diff
@@ -81,7 +83,12 @@ public void loadInstants(HoodieTableMetaClient metaClient,
                            HoodieArchivedTimeline.LoadMode loadMode,
                            Function<GenericRecord, Boolean> commitsFilter,
                            BiConsumer<String, GenericRecord> recordConsumer) {
-    Set<String> instantsInRange = new HashSet<>();
+    Set<String> instantsInRange = new HashSet<>();    
+    BoundedRecordConsumer boundedConsumer = recordConsumer instanceof BoundedRecordConsumer
+        ? (BoundedRecordConsumer) recordConsumer
+        : null;
```

Comment:
may be, its better to introduce UnBoundedConsumer. and so in L128, we can void != null check. the UnBoundedConsumer will always return false for `shouldStop()`